src/table/table-role/utils.ts

```diff
@@ -137,16 +124,32 @@ export function restoreTableFocusables(table: HTMLTableElement) {
   }
 }
 
-function isWidgetCell(cell: HTMLElement) {
-  return cell.getAttribute('data-widget-cell') === 'true';
+/**
+ * Returns true if the target element or one of its parents is a dialog or is marked with data-awsui-table-suppress-navigation.
+ * For dialog cells when in focus the tab indices are not overridden and keyboard events are not intercepted.
+ */
+function isDialogElement(target: HTMLElement) {
+  let current: null | HTMLElement = target;
+  while (current) {
+    const tagName = current.tagName.toLowerCase();
+    if (tagName === 'td' || tagName === 'th') {
+      return false;
+    }
+    if (
+      current.getAttribute('role') === 'dialog' ||
+      current.getAttribute('data-awsui-table-suppress-navigation') === 'true'
+    ) {
+      return true;
+    }
+    current = current.parentElement;
```

Comment:
Could we use the existing `findUpUntil` util for this?

Reply:
Technically, yes -- but that would be awkward due to the `tagName === 'td' || tagName === 'th'` check. The idea is to exit as soon the cell element is reached as checking further is not expected. Because the findUpUntil does not have any exit mechanism it would take an external boolean flag to check if the condition has been met.